src/Cake/Scripting/Roslyn/RoslynScriptSession.cs

```diff
@@ -30,6 +30,10 @@ internal sealed class RoslynScriptSession : IScriptSession
 
         public HashSet<string> Namespaces { get; }
 
+        public bool SupportsCachedExecution => true;
+
+        public bool IsCacheValid => System.IO.File.Exists(GetCachedAssemblyPath()) && !_options.ForceRecompile;
```

Comment:
This feels like very aggressive caching. Unless I'm missing something, if a newer version of Cake isn't installed, updates to the dev scripts will get ignored unless the explicit force recompile option is passed. How do we ensure updates take effect on a dev's machine?

Could we time how long the analysis step takes? It feels like a hash of the result from [this](https://github.com/hudl/cake/pull/1/files#diff-e025068a08bde0595593b8a1b690ed19R142) could be a more robust stage to cache at.

Reply:
yeah @jamiegs was just mentioning this. I think we can compute a hash of the files much like we do in CICD for output caching. I think the analyzer does take some time to run, getting a benchmark of it would be a good next step.